server/services/service.js

```diff
@@ -74,16 +74,23 @@ module.exports = () => ({
       if (config.drafts) return true;
       return x.publishedAt;
     });
-
-    return dataFiltered.map((x) => ({
-      id: x.id,
-      title: config.titleField ? x[config.titleField] : config.startField,
-      startDate: x[config.startField],
-      endDate: config.endField
-        ? x[config.endField]
-        : moment(x[config.startField]).add(config.defaultDuration, 'minutes'),
-      color: config.colorField ? x[config.colorField] : null,
-    }));
+    return dataFiltered.map((x) => {
```

Comment:
I think the way variables names are defined is a bit confusing.

I feel startDate & endDate should be as before considering all prev rules,
this should handle cases where the defaultDuration is 1440 minutes
```js
let startDate = x[config.startField];
let endDate = config.endField
         ? x[config.endField]
         : moment(x[config.startField]).add(config.defaultDuration, 'minutes');
```

with that out of the way, we can now check if it's an allDay event using [moment.diff](https://momentjs.com/docs/#/displaying/difference/) and adjust the variables if it's a 24h event
```js
const allDay = moment(endDate).diff(startDate, "hours") === 24;
if (addDay) {
 startDate = moment(startDate).startOf('day');
 endDate = moment(startDate).endOf('day');
}
```

this should handle cases where 
1. the default duration is 24 hours 
2. where the endDate is 24 hours after the start date

I guess my question would be, do we want to span across multiple days?
perhaps an even better solution is to have an `allDay` boolean as part of the content (this is how regular calendars work)

Reply:
Hi Luis,

Yes, renaming `isMultiDay` back to `allDay` aligns better with FullCalendar’s convention and expected behaviour. I think I was incorrectly too focused on the `allDay` field determining the desired result of my use case.. 

The `allDay` flag determines whether an event spans the entire day, while the startDate and endDate manage how long the event spans, including multi-day events, so that is easily managed in our logic.

So, for multi-day events, it's primarily the endDate that defines if an event spans multiple days, as when viewing in month view. The `allDay` flag can be used for full-day events (whether single or multi-day). If `allDay` is set to null, FullCalendar will rely on the default values, and if the `endDate` extends beyond the start date, the calendar will visually span across those days.

Multiple Days - this is definitely required for my use case, where I am designing a campsite booking application! 

I did not test your code, but I think this address your concerns while achieving my desired use case functionality:
   
```
   return dataFiltered.map((x) => {
      let startDate = x[config.startField];
      let endDate = config.endField
        ? x[config.endField]
        : moment(x[config.startField]).add(config.defaultDuration, 'minutes').format();

      // Check if the default duration equals 1440 minutes, meaning it's a full-day event
      const fullDay = config.defaultDuration === 1440;

      if (fullDay) {
        startDate = moment(startDate).startOf('day').format();
        endDate = moment(endDate).endOf('day').format();
      }

      return {
        id: x.id,
        title: config.titleField ? x[config.titleField] : x[config.startField],
        startDate: startDate,
        endDate: endDate,
        color: config.colorField ? x[config.colorField] : null,
        allDay: fullDay, // Full-day flag
      };
    });
```
Please see screen shots of working example.
![strapi-fullcalendar-allday-multiple-days-day-view](https://github.com/user-attachments/assets/1b373a4e-9ff3-4c5f-8469-ad930728d244)
![strapi-fullcalendar-allday-span-multiple-days-month-view](https://github.com/user-attachments/assets/90e6d68e-68ae-405d-b95b-87bc807a10e8)
![strapi-fullcalendar-allday-span-multiple-days-week-view](https://github.com/user-attachments/assets/c54286dc-690b-4b67-b89b-0db2741ee2fa)

Reply:
Thank  you Jacob, 
I agree with everything.

Please push your changes when you can :)